File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--changing-cached-data-in-a-workbook-on-a-server_1.cs
        private AdventureWorksDataSet.AdventureWorksLTDataSetTableAdapters.ProductTableAdapter productTableAdapter =
            new AdventureWorksDataSet.AdventureWorksLTDataSetTableAdapters.ProductTableAdapter();

        private void Sheet1_Startup(object sender, System.EventArgs e)
        {
            if (this.NeedsFill("adventureWorksLTDataSet"))
            {
                this.productTableAdapter.Fill(this.adventureWorksLTDataSet.Product);
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--changing-cached-data-in-a-workbook-on-a-server_1.vb
    Private ProductTableAdapter As New  _
        AdventureWorksDataSet.AdventureWorksLTDataSetTableAdapters.ProductTableAdapter()

    Private Sub Sheet1_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup
        If Me.NeedsFill("AdventureWorksLTDataSet") Then
            Me.ProductTableAdapter.Fill(Me.AdventureWorksLTDataSet.Product)
        End If
    End Sub